docs/design/metadata.md

```diff
@@ -0,0 +1,365 @@
+  unconditionally.
+
+
+## `pyproject.toml` file
+
+### Base information
+
+The `pyproject.toml` file is the standard project configuration file
+as defined in [pyproject.toml specification](
+https://packaging.python.org/en/latest/specifications/pyproject-toml/#pyproject-toml-spec).
+Variant-enabled packages utilize a dedicated top-level `variant` table.
+
+### Example contents
+
+```toml
+[variant.default-priorities]
+namespace = ["x86_64", "aarch64"]
+feature = ["x86_64::level"]
+property = ["x86_64::avx2::on"]
+
+[variant.providers.aarch64]
+requires = [
+    "provider-variant-aarch64 >=0.0.1,<1; python_version >= '3.9'",
+    "legacy-provider-variant-aarch64 >=0.0.1,<1; python_version < '3.9'",
+]
+enable-if = "platform_machine == 'aarch64' or 'arm' in platform_machine"
+plugin-api = "provider_variant_aarch64.plugin:AArch64Plugin"
+
+[variant.providers.x86_64]
+requires = ["provider-variant-x86-64 >=0.0.1,<1"]
+enable-if = "platform_machine == 'x86_64' or platform_machine == 'AMD64'"
+plugin-api = "provider_variant_x86_64.plugin:X8664Plugin"
+```
+
+### `default-priorities` subtable
+
+The `variant.default-priorities` table contains the [default prority
+information](#default-priorities). It may contain the following keys:
+
+- `namespace` specifying the default namespace priority list. This is
+  a list of strings, where every string must be a namespace name.
+  As indicated in the common metadata section, this key is obligatory.
+
+- `feature` specifying the default feature priority list. This is a list
+  of the canonical feature strings (`namespace :: feature`). This key
+  is optional.
+
+- `property` specifying the default property priority list. This is
+  a list of the canonical property strings
+  (`namespace :: feature :: value`). This key is optional.
+
+### `providers` subtable
+
+The `variant.providers` table contains the [provider information](
+#provider-information). It contains subtables keyed after namespaces,
+and must contain a subtable for every namespace specified
+in `variant.default-priorities.namespace` list.
+
+Every subtable may contain the following keys:
+
+- `plugin-api` that is a string forming the object reference used
+  to load the plugin. It is required.
+
+- `requires` that is a list of strings, where every string is
+  a dependency specifier expressing how to install the provider plugin.
+  It is optional, but it is necessary to support installing plugins
+  in an isolated environment.
+
+- `enable-if` that is a string forming an environment marker,
+  specifying the condition under which the plugin is enabled. It is
+  optional, and the plugin is always enabled if it missing.
+
+
+## Distribution metadata
+
+### Base information
+
+The distribution metadata is defined in the [core metadata
+specifications](
+https://packaging.python.org/en/latest/specifications/core-metadata/).
+The wheel variant format adds headers corresponding to the common
+metadata and to the wheel variant description.
```

Comment:
Put my review comment into a separate issue for discussion: https://github.com/wheelnext/pep_xxx_wheel_variants/issues/44

Reply:
Thanks. I'm going to mark this comment as resolved -- the purpose of this PR is really to document how things are (and of the review to make sure that the documentation is clear), rather than changing them right now.